### ISSUE TITLE
ChangeLogWriter cannot handle more than 500 logs

### ISSUE THREAD

**Issue by following5** [opencaching/okapi#376]
A new field has been added to the OC.de cache_logs table and then initialized. This has triggered an OKAPI replication check of all log entries, though the new field has no effect onto the log entries as presented by OKAPI.
```
===== ERROR MESSAGE =====
Parameter 'log_uuids' has invalid value: Maximum allowed number of referenced log entries is 500. You provided 9260 UUIDs.
=========================

--- Stack trace ---
#0 [internal function]: okapi\services\logs\entries\WebService::call(Object(okapi\OkapiInternalRequest))
#1 /var/www/opencaching.de/www/code/htdocs/okapi/service_runner.php(136): call_user_func(Array, Object(okapi\OkapiInternalRequest))
#2 /var/www/opencaching.de/www/code/htdocs/okapi/services/replicate/replicate_common.inc.php(293): okapi\OkapiServiceRunner::call('services/logs/e...', Object(okapi\OkapiInternalRequest))
#3 /var/www/opencaching.de/www/code/htdocs/okapi/services/replicate/replicate_common.inc.php(139): okapi\services\replicate\ReplicateCommon::generate_changelog_entries('services/logs/e...', 'log', 'log_uuids', 'uuid', Array, 'uuid|cache_code...', false, true, 3600)
#4 /var/www/opencaching.de/www/code/htdocs/okapi/cronjobs.php(551): okapi\services\replicate\ReplicateCommon::update_clog_table()
#5 /var/www/opencaching.de/www/code/htdocs/okapi/cronjobs.php(92): okapi\cronjobs\ChangeLogWriterJob->execute()
#6 /var/www/opencaching.de/www/code/htdocs/okapi/core.php(1242): okapi\cronjobs\CronJobController::run_jobs('cron-5')
#7 /var/www/opencaching.de/www/code/htdocs/okapi/views/cron5.php(36): okapi\Okapi::execute_cron5_cronjobs()
#8 [internal function]: okapi\views\cron5\View::call()
#9 /var/www/opencaching.de/www/code/htdocs/okapi/controller.php(77): call_user_func_array(Array, Array)
#10 /var/www/opencaching.de/www/code/htdocs/okapi/controller.php(98): okapi\OkapiScriptEntryPointController::dispatch_request('/okapi/cron5')
#11 {main}

--- OKAPI method called ---
/okapi/cron5

--- OKAPI version ---
1154 (f24c7ac20d97d4e9466c58ba2333b0d324cb2e0a)

```

What can I do to fix this? I will disable the OKAPI cronjob now.

**Comment by wrygiel** [comment]
I think it should be possible to use the `skip_limits` flag to fix this error. However, I'm not sure if it has been implemented in the `logentries` method.

**Comment by following5** [comment]
Ok, `skip_limìts` should work.

**Comment by wrygiel** [comment]
If you're sure that nothing changed, you could also try changing `okapi_syncbase` by force. My guess is that the value of `okapi_syncbase` was bumped up by your change somehow.

**Comment by wrygiel** [comment]
If indeed all log entries are now marked for checking, then using `skip_limits` may cause OKAPI to try to fetch **all** of them to memory. That would just cause major swapping and kill your server... But if this would be the case, then why would the error say that only 9260 log entries are involved? I guess you'll have to debug the code more thoroughly, I don't remember exactly. :(

**Comment by following5** [comment]
When processing log entries, the replication mechanism splits them down to chunks of 100 logs - no problem there.

The problem was the 9260 deleted logs in the `cache_logs_archived` table, which was modified the same way. OKAPI processes them all in one.

**Comment by wrygiel** [comment]
Aaah, yes. Forgot you store them in separate table. Will you fix this or should I?

**Comment by following5** [comment]
fixed by commit 7991fedc
